### PR TITLE
{lib}[GCCcore/11.3.0] UCC-CUDA v1.0.0

### PR DESCRIPTION
**Pull request by bartoldeman**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by bartoldeman** [comment]
I used symlinks to avoid needing to patch the non-CUDA UCC. That's simple but a little different from the way UCX-CUDA does things.

**Comment by Micket** [comment]
> I used symlinks to avoid needing to patch the non-CUDA UCC. That's simple but a little different from the way UCX-CUDA does things.

So, this would possible become and issue if we needed more UCC-xxx packages for additional plugins. I can only imagine that being the case for ROCM, which case you wouldn't have UCC loaded simultaneously

**Comment by bartoldeman** [comment]
UCC has no support for ROCM as far as I can see.
Configure only looks for NCCL, CUDA, UCX and SHARP, but SHARP is more os-level (IB related).

**Comment by bartoldeman** [comment]
@boegelbot please test @ generoso

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
build-node.computecanada.ca - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/08367ad51aad03490e301434f2cfbb6e for a full test report.

File: easybuild/easyconfigs/u/UCC-CUDA/UCC-CUDA-1.0.0-GCCcore-11.3.0-CUDA-11.7.0.eb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+builddependencies = [
+    ('binutils', '2.38'),
+    ('Autotools', '20220317'),
+]
+
+dependencies = [
+    ('UCC', '1.0.0'),
+    ('CUDA',  '11.7.0', '', True),
+    ('NCCL', '2.12.12', '-CUDA-%(cudaver)s'),
+]
```
Comment (Micket) [inline easybuild/easyconfigs/u/UCC-CUDA/UCC-CUDA-1.0.0-GCCcore-11.3.0-CUDA-11.7.0.eb:38]: Perhaps add a direct dep on UCX-CUDA as well?

Comment (bartoldeman) [reply easybuild/easyconfigs/u/UCC-CUDA/UCC-CUDA-1.0.0-GCCcore-11.3.0-CUDA-11.7.0.eb]: done

**Comment by bartoldeman** [comment]
@boegelbot please test @ generoso

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.5, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/cdd81c1b367aa4d9d15a5aa3d857644d for a full test report.

**Comment by Micket** [comment]
Going in, thanks @bartoldeman!

**Pull request merged**